gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts

```diff
@@ -16,22 +16,81 @@
 import { Component, Inject } from '@angular/core';
 import { MatButtonModule } from '@angular/material/button';
 import { MAT_DIALOG_DATA, MatDialogModule, MatDialogRef } from '@angular/material/dialog';
+import { MatFormField, MatLabel } from '@angular/material/form-field';
+import { MatInput } from '@angular/material/input';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatButtonToggleModule } from '@angular/material/button-toggle';
+import { CommonModule } from '@angular/common';
+import { GioFormFocusInvalidModule } from '@gravitee/ui-particles-angular';
+import { map, startWith } from 'rxjs/operators';
+import { Observable } from 'rxjs';
 
-export interface EnvironmentFlowsAddEditDialogData {}
+import { ApiV4, ExecutionPhase } from '../../../../entities/management-api-v2';
 
-export type EnvironmentFlowsAddEditDialogResult = boolean;
+export interface EnvironmentFlowsAddEditDialogData {
+  apiType: ApiV4['type'];
+}
+
+export type EnvironmentFlowsAddEditDialogResult = undefined | { name: string; description?: string; phase: ExecutionPhase };
+
+const PHASE_BY_API_TYPE: Record<ApiV4['type'], ExecutionPhase[]> = {
+  PROXY: ['REQUEST', 'RESPONSE'],
+  MESSAGE: ['REQUEST', 'MESSAGE_REQUEST', 'MESSAGE_RESPONSE', 'RESPONSE'],
```

Comment:
Maybe in this order ?
['REQUEST', 'RESPONSE', 'MESSAGE_REQUEST', 'MESSAGE_RESPONSE']
Or this one ?
['REQUEST', 'MESSAGE_REQUEST', 'RESPONSE', 'MESSAGE_RESPONSE']

Reply:
I originally had the 1st order . but i changed it for some reason haha. there's no case where the response executes at the end of the message exchange. but that must depend on your endpoint. 

The both suits me. as you prefer ?